engine.py

```diff
@@ -521,13 +521,23 @@ def _run_commands_at_startup(self):
         # Set the _callback_from_non_pane_menu hint so that the show_panel method knows this
         # was invoked not from the pane menu.
 
-        # FIXME: This pattern is horrible.
-        setattr(tank, "_callback_from_non_pane_menu", True)
-        try:
-            for command in commands_to_run:
-                command()
-        finally:
-            delattr(tank, "_callback_from_non_pane_menu")
+        def run_at_startup():
+            # FIXME: This pattern is horrible.
+            tank._callback_from_non_pane_menu = True
```

Comment:
I change this line because the dog asked me to. Otherwise it's essentially the same code as before, but called from the single shot timer.